arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedReaderBuilder.java

```diff
@@ -61,6 +61,10 @@ public VectorizedReaderBuilder(
     this.readerFactory = readerFactory;
   }
 
+  public Function<List<VectorizedReader<?>>, VectorizedReader<?>> readerFactory() {
```

Comment:
I think we may not this method. The second method should be enough if `ReadBuilder` is changed.

```
@Override
protected VectorizedReader<?> vectorizedReader(List<VectorizedReader<?>> reorderedFields) {
  VectorizedReader<?> reader = super.vectorizedReader(reorderedFields);
  if (deleteFilter != null) {
    ((ColumnarBatchReader) reader).setDeleteFilter(deleteFilter);
  }
  return reader;
}
```

Reply:
Nice suggestion!